prisma/schema.prisma

```diff
@@ -1947,6 +1998,163 @@ model HomeBlock {
+  entryMode               BountyEntryMode @default(Open)
+  type                    BountyType
+  minBenefactorUnitAmount Int
+  maxBenefactorUnitAmount Int? // Default to initial benefactor's entry
+  entryLimit              Int             @default(1)
+  nsfw                    Boolean         @default(false)
+  poi                     Boolean         @default(false)
+  complete                Boolean         @default(false)
+
+  tags        TagsOnBounty[]
+  entries     BountyEntry[]
+  benefactors BountyBenefactor[]
+  engagements BountyEngagement[]
+  thread      Thread?
+  metrics     BountyMetric[]
+  rank        BountyRank?
+  stats       BountyStat?
+  reports     BountyReport[]
+}
+
+model BountyEntry {
+  id        Int      @id @default(autoincrement())
+  userId    Int?
+  user      User?    @relation(fields: [userId], references: [id], onDelete: SetNull)
+  bountyId  Int
+  bounty    Bounty   @relation(fields: [bountyId], references: [id], onDelete: Cascade)
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+  locked    Boolean  @default(false)
+
+  benefactors BountyBenefactor[]
+  thread      Thread?
+  reactions   BountyEntryReaction[]
+  metrics     BountyEntryMetric[]
+  rank        BountyEntryRank?
+  stats       BountyEntryStat?
+  reports     BountyEntryReport[]
+}
+
+model BountyEntryReaction {
+  bountyEntry   BountyEntry     @relation(fields: [bountyEntryId], references: [id], onDelete: Cascade)
+  bountyEntryId Int
+  user          User            @relation(fields: [userId], references: [id], onDelete: Cascade)
+  userId        Int
+  reaction      ReviewReactions
+  createdAt     DateTime        @default(now())
+
+  @@id([bountyEntryId, userId, reaction])
+}
+
+model BountyBenefactor {
+  userId      Int
+  user        User         @relation(fields: [userId], references: [id], onDelete: Cascade)
+  bountyId    Int
+  bounty      Bounty       @relation(fields: [bountyId], references: [id], onDelete: Cascade)
+  unitAmount  Int
+  createdAt   DateTime     @default(now())
+  updatedAt   DateTime     @updatedAt
+  awardedAt   DateTime?
+  awardedToId Int?
+  awartedTo   BountyEntry? @relation(fields: [awardedToId], references: [id], onDelete: SetNull)
+  currency    Currency     @default(BUZZ)
+
+  @@id([userId, bountyId])
+}
+
+enum BountyEngagementType {
+  Favorite
+  Track
+}
+
+model BountyEngagement {
+  userId    Int
+  user      User                 @relation(fields: [userId], references: [id], onDelete: Cascade)
+  bountyId  Int
+  bounty    Bounty               @relation(fields: [bountyId], references: [id], onDelete: Cascade)
+  type      BountyEngagementType
+  createdAt DateTime             @default(now())
+
+  @@id([userId, bountyId, type])
+  @@index([bountyId])
+}
+
+model TipConnection {
+  transactionId String // guid
+  entityId      Int
+  entityType    String
+
+  @@id([entityId, entityType, transactionId])
+}
+
+model BountyMetric {
```

Comment:
Consider indexes on the individual metrics for faster lookups at the expense of heavier computations.  (same for BountyEntryMetric)

Reply:
We don't actually use BountyMetric for querying/sorting, we do it with the Rank views, do we still want to index those?

Reply:
If we don't filter / sort on them then no, lets not index them